### DISCUSSION TITLE
Add user from remote client

### DISCUSSION THREAD

**Discussion by PalGiupo** [Kozea/Radicale#1603] [Q&A server]
Hi, I created a Radicale server and I added users from the terminal. Is not there a way to create users remotely? I need to share the server among more friends, and they must be able to create their users remotely. I read the guide but I did not find anything about it. Could not you tell me if there is a way to do it? 
The login is for registered users only, but I wanted a way to add new ones.
Thanks

**Comment by jonassmedegaard** [comment]
Quoting PalGiupo (2019-03-10 15:01:49)
> Hi, I created a Radicale server and I added users from the terminal. Is not there a way to create users remotely? I need to share the server among more friends, and they must be able to create their users remotely. I read the guide but I did not find anything about it. Could not you tell me if there is a way to do it? Thanks

The CalDAV and CardDAV protocols does not include language to describe 
"create this user with that password".

You can setup one account with one password, and then share that one 
account + password among all your friends, and everyone can write on all 
calendars and all contactbooks.

You can setup a separate web application independently of Radicale which 
provides some way of asking for username and password, and creates 
accounts (either failing or overwriting if choosing an existing name).
You can then tell your friends to first visit that other we page, and 
when done fill in same information in their CalDAV/CardDAV clients.

Or you can do same with an sms app, or an irc bot, or whatever other 
interface you can think of...


 - Jonas

-- 
 * Jonas Smedegaard - idealist & Internet-arkitekt
 * Tlf.: +45 40843136  Website: http://dr.jones.dk/

 [x] quote me freely  [ ] ask before reusing  [ ] keep private

**Comment by Tntdruid** [comment]
Setup a mailserver and use imap auth plugin

**Comment by PalGiupo** [comment]
And if I created a Flask front-end to create LDAP users, and I have Radicale users authenticated on LDAP, could it work? it's a good idea? Or is it too complicated? I've never used LDAP, but it seems useful
